govcd/vm.go

```diff
@@ -456,3 +457,68 @@ func (vm *VM) Undeploy() (Task, error) {
 	return *task, nil
 
 }
+
+// Attach or detach an independent disk
+// Use the disk/action/attach or disk/action/detach links in a Vm to attach or detach an independent disk.
+// Reference: vCloud API Programming Guide for Service Providers vCloud API 30.0 PDF Page 164 - 165,
+// https://vdc-download.vmware.com/vmwb-repository/dcr-public/1b6cf07d-adb3-4dba-8c47-9c1c92b04857/
+// 241956dd-e128-4fcc-8131-bf66e1edd895/vcloud_sp_api_guide_30_0.pdf
+func (vm *VM) attachOrDetachDisk(diskParams *types.DiskAttachOrDetachParams, rel string) (Task, error) {
+	var err error
+	var execLink *types.Link
+	for _, link := range vm.VM.Link {
+		if link.Rel == rel && link.Type == types.MimeDiskAttachOrDetachParams {
```

Comment:
Is this verified? - that attach API call can be used to detach?

Reply:
Yes, you can find the details in vCloud API Document.